tests/test_uncrustify/test.py

```diff
@@ -108,18 +112,30 @@ def run(self, args):
         except subprocess.CalledProcessError as exc:
             msg = '{}: Uncrustify error code {}'
             msg = msg.format(self.test_name, exc.returncode)
-            printc('FAILED: ', msg, **FAIL_ATTRS)
-            raise ExecutionFailure(exc)
+            if not self.test_xfail:
```

Comment:
I was wondering, earlier, if we need to distinguish between tests that are supposed to produce bad output vs. tests that cause uncrustify to *fail*... I think, for cases where failing is the *correct* behavior, we should be using front-end tests, so any failure of a *staging* test is a known bug. That being the case, I am unconvinced that (always) printing a message for these cases is TRTTD; it should just be treated as a pass (IOW, tweak `run_tests` to print either `PASSED` or `XFAILED`, but here, just `return`.)

Reply:
yeah, hard to get it right. If we know that it should fail then nothing should be printed but uncrustify does prints to console by default (we're not calling with `check_output`). But then if we do that and want to capture everything then you lose the stdout/stderr interleaving which probably can be acceptable. But till that happens, I went with printing also a message to explain why something was showing as a failure but it's passing because it's an expected failure.

Reply:
> but uncrustify does prints to console by default

Hmm... okay, I see the "problem"... This does make me inclined to think we should capture stdout/stderr for an xfail test (or maybe even all tests?) and suppress it. At least, that would be the way to implement my knee-jerk preference, which is to continue to have no output unless a test fails or `-p` is given.

Presumably, `stdout=subprocess.PIPE, stderr=subprocess.STDOUT` would preserve interleaving? (OTOH, do we care about redirecting the handles of `run_tests.py` having 1:1 mapping to redirecting uncrustify's handles?)

Reply:
> unless a test fails or -p is givenI agree with that

Reply:
Ah... well, I'm going to be fiddling in here soon-ish anyway (actually I have a WIP to integrate the front-end tests). If you remind me, I can probably do that conversions while I'm at it :slightly_smiling_face:.

Reply:
Yeah, I'm unsure. In the last update everything looks like it's working, even for debug output which is a lot of text.